bigchaindb/upsert_validator/validator_election.py

```diff
@@ -141,3 +146,86 @@ def create(cls, tx_signers, recipients, metadata=None, asset=None):
+                    for output in txn['outputs']:
+                        # NOTE: We enforce that a valid vote to election id will have only
+                        # election_pk in the output public keys, including any other public key
+                        # along with election_pk will lead to vote being not considered valid.
+                        if len(output['public_keys']) == 1 and [election_pk] == output['public_keys']:
+                            votes = votes + int(output['amount'])
+            else:
+                if txn.operation == 'VALIDATOR_ELECTION_VOTE':
+                    for output in txn.outputs:
+                        # NOTE: We enforce that a valid vote to election id will have only
+                        # election_pk in the output public keys, including any other public key
+                        # along with election_pk will lead to vote being not considered valid.
+                        if len(output.public_keys) == 1 and [election_pk] == output.public_keys:
+                            votes = votes + int(output.amount)
+        return votes
+
+    def get_commited_votes(self, bigchain, election_pk=None):
+        if election_pk is None:
+            election_pk = self.to_public_key(self.id)
+        txns = list(backend.query.get_asset_tokens_for_public_keys(bigchain.connection,
+                                                                   self.id,
+                                                                   [election_pk],
+                                                                   'VALIDATOR_ELECTION_VOTE'))
+        return self.count_votes(election_pk, txns)
+
+    @classmethod
+    def conclude(cls, bigchain, txn_id, current_votes=[], height=None):
+        """Check if the given election has concluded or not
+        NOTE:
+        * Election is concluded iff the current validator set is exactly equal
+          to the validator set encoded in election outputs
+        * Election can concluded only if the current votes achieves a supermajority
+        """
+        election = bigchain.get_transaction(txn_id)
+
+        if election:
+            election_pk = election.to_public_key(election.id)
+            votes_commited = election.get_commited_votes(bigchain, election_pk)
+            votes_current = election.count_votes(election_pk, current_votes)
+            current_validators = election.current_validators(bigchain, height)
+
+            if election.is_same_topology(current_validators, election.outputs):
+                total_votes = sum(current_validators.values())
+                if (votes_commited < (2/3)*total_votes) and \
+                   (votes_commited + votes_current > (2/3)*total_votes):
+                    return election
+
+        return False
+
+    @classmethod
+    def get_validator_update(cls, bigchain, new_height, txns):
+        votes = {}
+        for txn in txns:
+            if isinstance(txn, ValidatorElectionVote):
```

Comment:
A neatpick - it might be slightly easier to read the code that avoids nesting by reversing the condition checks.

```
for txn in txns:
    if not isintance .. :
        continue
    ....
    if not election:
        continue
    ....
```